### PR TITLE
Feat/add sorting tasks page

### LINKED ISSUE #1470: add sorting to the tasks list
as a task user
with admin permissions or higher
when i view the list of tasks
then i would like to be able to sort this list by any column

### PR DESCRIPTION
**Pull request by Palakp41**
Closes #1470 

_What was the solution?_ Added sorting on tasks header to sort the tasks row.. Also, created tests for tasks list.

  - [ ] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [ ] Rebased/mergeable
  - [ ] Tests pass
  - [ ] swagger.json updated (if modified Go structs or API)
  - [ ] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: ui/src/tasks/components/TasksList.tsx [hunk 1]
```diff
@@ -42,23 +92,37 @@ export default class TasksList extends PureComponent<Props> {
...
-    return tasks.map(t => (
-      <TaskRow
-        key={`task-id--${t.id}`}
-        task={t}
-        onActivate={onActivate}
-        onDelete={onDelete}
-        onSelect={onSelect}
-      />
-    ))
+  private rows = (tasks: Task[]): JSX.Element => {
```
[the remark this answers is not in the document]
Comment (Palakp41) [reply ui/src/tasks/components/TasksList.tsx]: no, This is the function that sortingHat accepts to be able to sort tasks and return the corresponding sorted JSX.Elements.

File: ui/src/dashboards/components/dashboard_index/Table.tsx [hunk 2]
```diff
@@ -85,49 +87,35 @@ class DashboardsTable extends PureComponent<Props & WithRouterProps, State> {
...
+    this.setState({sortKey, sortDirection: nextSort})
   }
 
-  private get defaultColumns(): Column[] {
+  private get columns(): IndexHeaderCellProps[] {
+    const {sortKey, sortDirection} = this.state
+    const headerKeys: SortKey[] = ['name', 'owner', 'modified', 'default']
     return [
       {
         columnName: 'name',
```
[the remark this answers is not in the document]
Comment (ebb-tide) [reply ui/src/dashboards/components/dashboard_index/Table.tsx]: They happen to be the same here, but are not guaranteed to be.

[the remark this answers is not in the document]
Comment (Palakp41) [reply ui/src/dashboards/components/dashboard_index/Table.tsx]: Thanks Daniel! fixed it.

File: ui/src/tasks/components/TasksList.tsx [hunk 3]
```diff
@@ -23,17 +26,64 @@ interface Props {
...
-    const {searchTerm, onCreate} = this.props
+type SortKey = keyof Task | 'organization.name'
+
+interface State {
+  sortKey: SortKey
+  sortDirection: Sort
+}
 
+export default class TasksList extends PureComponent<Props, State> {
+  public headerKeys: SortKey[] = [
```
[the remark this answers is not in the document]
Comment (Palakp41) [reply ui/src/tasks/components/TasksList.tsx]: Thanks, fixed it!

File: ui/src/tasks/components/TasksList.tsx [hunk 4]
```diff
@@ -23,17 +26,64 @@ interface Props {
...
+    const {sortKey, sortDirection} = this.state
+    const headerKeys = this.headerKeys
     return (
       <IndexList>
         <IndexList.Header>
-          <IndexList.HeaderCell columnName="NAME" width="20%" />
-          <IndexList.HeaderCell columnName="ACTIVE" width="10%" />
-          <IndexList.HeaderCell columnName="SCHEDULE" width="20%" />
-          <IndexList.HeaderCell columnName="ORGANIZATION" width="15%" />
+          <IndexList.HeaderCell
```
[the remark this answers is not in the document]
Comment (Palakp41) [reply ui/src/tasks/components/TasksList.tsx]: Will consider it!

**Pull request merged**